src/fluree/db/connection.cljc

```diff
@@ -434,6 +432,25 @@
   (remove-listener* (:state conn) network dbid key))
 
 
+
+(defn add-token
+  "Adds token to connection information so it is available to submit storage read requests.
+
+  Returns true if successful, false otherwise."
+  [conn token]
+  (let [conn-id (:id conn)
+        _ (log/info "add-token" {:conn-id conn-id :token token})
+        ]
+    (try
+      (swap! server-connections-atom update-in [conn-id :token]
+             (fn [x]
+               (if x
+                 x
+                 token)))
+      true
+      (catch #?(:clj Exception :cljs :default) _
```

Comment:
Can use our new try*/catch* macro that automatically resolves the :clj/:cljs conditions so you don't need to do that here.